spec/example_app/app/controllers/admin/customers_controller.rb

```diff
@@ -2,6 +2,14 @@ class Admin::CustomersController < Admin::ApplicationController
   private
 
   def scoped_resource
-    Customer.where(hidden: false)
+    super.where(hidden: false)
+  end
+
+  def resource_class_for_action(action_name)
+    if action_name.to_s == "index"
+      Customer::Index
+    else
+      super
+    end
```

Comment:
a-ha. Here is the missing piece. Looks good.

A second approach could be placing something in `CustomersDashboard`, maybe like:

```ruby
action_classes = {
  index: Customer::Index
}
```

...which would leave less for the developer to worry about, because all of the logic is in the dashboard class.

Maybe; I've been gone a while so I'm unsure if the second approach is sound.

Reply:
That's a good point. I'm still unsure though, of where Administrate should be adding dashboard options and where it shouldn't. In general, the feature introduced in this PR feels to me like something that should be supported, but that in reality won't be used that often. For this reason, it doesn't necessarily deserve more explicit configuration options, as these become a maintenance burden later for little benefit.

It may still be a good idea, but the benefit of starting the feature at a controller level is that we can delay adding new dashboard-level options later.

Reply:
This makes sense. Adding action-specific models seems like a niche usage, so overriding the controller is a good approach.